<?xml version="1.0"?>

<launch>

  <!-- arguments -->
  <!-- <arg name="world_frame" default="/world" /> -->
<!--   <arg name="Ds" value="cf4,cf5"/>
  <arg name="Is" value="cf0" /> -->
  <arg name="uri" default="radio://0/84/1M/E7E7E7E703" />
  <arg name="frame" default="cf4" />
  <arg name="logger_dir" default="res1" />
<!--   <arg name="id" default="D1"/> -->
<!--   <arg name="v" default=".3"/>
  <arg name="r" default=".5"/>
  <arg name="z" default=".5"/>
  <arg name="r_close" value="1.2">
  <arg name="k_close" value="0.9"> -->

  <!-- the intruder crazyflie -->
  <group ns='$(arg frame)'>
    <!-- add crazyflie -->
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri)" />
      <arg name="tf_prefix" value="$(arg frame)" />
      <arg name="enable_logging" value="True" />
    </include>

    <!-- add inner-loop controller -->
    <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
      <arg name="frame" value="$(arg frame)" />
    </include>
    <!-- add outer-loop controller, hover only for now -->
<!--     <include file="$(find crazyflie_game)/launch/strategy.launch">
      <arg name="world_frame" value="$(arg world_frame)" />
      <arg name="cf_frame" value="$(arg frame)" />
      <arg name="Ds" value="$(arg Ds)" />
      <arg name="Is" value="$(arg Is)" />
      <arg name="player_id" value="$(arg id)"/>
      <arg name="velocity" value="$(arg v)"/>
      <arg name="cap_range" value="$(arg r)"/>
      <arg name="r_close" value="$(arg r_close)" />
      <arg name="k_close" value="$(arg k_close)" />
      <arg name="speed_ratio" value="$(arg a)"/>
      <arg name="z" value="$(arg z)"/>
    </include> -->
    
    <!-- add data recorder for each player -->
    <include file="$(find crazyflie_game)/launch/player_recorder.launch">
      <arg name="cf_frame" value="/$(arg frame)" />
      <arg name="logger_dir" value="$(arg logger_dir)" />
    </include>

  </group>

</launch>